core/base/inc/Rtypes.h

```diff
@@ -267,7 +267,8 @@ class ClassDefGenerateInitInstanceLocalInjector:
 // DeclFileLine() is not part of it since CINT uses that as trigger for
 // the class comment string.
 #define _ClassDefBase_(name, id, virtual_keyword, overrd)                                                       \
-private:                                                                                                        \
+private:          \
+   static_assert(std::is_integral<decltype(id)>::value, "ClassDef(Inline) macro: the specified ID is not an integer.");                                                        \
```

Comment:
`ID` -> `version number`?

Reply:
Do we have the relevant header for `is_integral` always `#include`d?

Reply:
nice suggestion to make the message even clearer.
For the header, probably not, it's just working but I will include type_traits to be more correct.